tests/agent/test_protocol.py

```diff
@@ -105,7 +106,84 @@ async def test_apply_from_client_when_units_with_output_and_updating_config(self
 
         await checkin_handler.apply_from_client()
 
-        assert config_wrapper_mock.try_update.called_once_with(unit_mock.config.source)
+        assert config_wrapper_mock.try_update.called_once_with(unit_mock)
+        assert service_manager_mock.restart.called
+
+    @pytest.mark.asyncio
+    async def test_apply_from_client_when_units_with_multiple_outputs_and_updating_config(
+        self,
+    ):
+        client_mock = Mock()
+        config_wrapper_mock = Mock()
+
+        config_wrapper_mock.try_update.return_value = True
+
+        service_manager_mock = Mock()
+
+        unit_es = Mock()
+        unit_es.unit_type = proto.UnitType.OUTPUT
+        unit_es.config.type = "elasticsearch"
+        unit_es.config.id = "config-es"
+
+        unit_kafka = Mock()
+        unit_kafka.unit_type = proto.UnitType.OUTPUT
+        unit_kafka.config.type = "kafka"
+        unit_kafka.config.id = "config-kafka"
+
+        client_mock.units = [unit_kafka, unit_es]
+
+        checkin_handler = ConnectorCheckinHandler(
+            client_mock, config_wrapper_mock, service_manager_mock
+        )
+
+        await checkin_handler.apply_from_client()
+
+        # Only ES output from the policy should be used by connectors component
+        assert config_wrapper_mock.try_update.called_once()
+        called_unit = config_wrapper_mock.try_update.call_args[0][0]
```

Comment:
We need to explicitly check the call arguments to compare IDs. The called_once_with assertion is more lenient and doesn’t verify object properties like IDs.